src/libponyc/ast/symtab.h

```diff
@@ -16,6 +16,7 @@ typedef enum
   SYM_DEFINED,
   SYM_UNDEFINED,
   SYM_CONSUMED,
+  SYM_CONSUMED_SAME_EXPR,
```

Comment:
I think this new case also needs to be accounted for in another place. You covered duplicating many of the uses of `SYM_CONSUMED`, but you also need to cover this one:
- `ast_all_consumes_in_scope` function in `ast.c`

Reply:
Thanks for the catch. Added in latest commit.